pyglet/gui/frame.py

```diff
@@ -87,49 +88,84 @@ def remove_widget(self, widget):
 
     def on_mouse_press(self, x, y, buttons, modifiers):
         """Pass the event to any widgets within range of the mouse"""
+
+        if self._active_entry: 
+            self._active_entry.on_mouse_press(x, y, buttons, modifiers)
+            self._active_entry = self._active_entry if self._active_entry._focus else None
+        
+
         for widget in self._cells.get(self._hash(x, y), set()):
-            widget.on_mouse_press(x, y, buttons, modifiers)
+            if type(widget).__name__ == 'TextEntry':
+                if self._active_entry == widget: continue
```

Comment:
Single line if blocks are very unusual in Python, and they don't seem to be used anywhere in pyglet. Maybe use black for automatic formatting.